core/domain/user_services.py

```diff
@@ -1880,6 +1884,22 @@ def mark_user_for_deletion(user_id):
         _save_user_identifiers(user_identifiers)
 
 
+def save_deleted_username(normalized_username):
+    """Save the username of deleted user.
+
+    Args:
+        normalized_username: str. Normalized version of the username to be
+            saved.
+    """
+    hashed_normalized_username = utils.convert_to_hash(
```

Comment:
Just to validate that the hash function we're using here is good -- can you please add an audit job that runs over all the current models, converts them to hashes, and counts frequency? If all are unique, then everything is OK, but this is the sort of thing we should check.

Reply:
Done.

Reply:
Please file a job run request?

Reply:
Just going to (I were checking the errors if they are not related to the job).